doc/api/util.md

```diff
@@ -946,6 +946,161 @@ Otherwise, returns `false`.
 See [`assert.deepStrictEqual()`][] for more information about deep strict
 equality.
 
+## `util.parseArgs([argv][, options])`
+<!-- YAML
+added: REPLACEME
+-->
+
+* `argv` {string[]|Object} (Optional) Array of argument strings; defaults
```

Comment:
Nit:
```suggestion
* `argv` {string[]} (Optional) Array of argument strings; defaults
```
since this argument cannot be an object itself, it is only an object if it is `options`. I don't think that we include 'replacement/nested/optional' types in our docs in such cases. IMO it will only confuse the reader as it removes clear type-boundary between optional arguments.

Reply:
would prefer to leave as-is unless the convention is explicitly documented somewhere.  afaik doing it the suggested way does not work very well if your docstrings are read by machines (e.g., TS language server), which is why I wrote it the way I did

Reply:
I prefer to keep this only being an array of strings. I do not understand the need for supporting anything that is not an array of strings given that `process.argv` is an array of strings. Can you please articulate why this is needed?

---

From experience adding types overloads are a cause issues down the line with the `Buffer` constructor being the most notable case of this. This is also very present in the streams API and other places, and I would define most of them as problematic one way or another.

Reply:
@mcollina It allows someone to supply an explicit array of arguments, or just an options object, where the value would default to process.argv. we could make it so you must pass process.argv, but that is likely boilerplate in 90% of cases.

Reply:
btw this comment is in regards to a docstring, unless you are talking about something else

Reply:
Sorry, I misunderstood this. `Object` should be removed from this line as it is confusly. We normally mark this parameter as optional and that's it.

Reply:
This still needs to be fixed.

Reply:
This seems resolved? Can we mark as resolved?